### PR TITLE
Allow users to edit grocery item

### PR DESCRIPTION
**Pull request by indirap**
* Also add swipe gestures for grocery item

### CODE REVIEW

File: src/Grocery/Grocery.js [hunk 1]
```diff
@@ -1,34 +1,131 @@
...
+			listId: props.navigation.state.params.listId,
+			name,
+			amount,
+			newGrocery
+		}
 	}
+    
+	addGrocery() {
+		if (this.state.name.length === 0) {
+			Alert.alert(
```
Comment (indirap) [inline src/Grocery/Grocery.js]: Not sure if it's just on my Mac, but the Alert makes the iOS simulator super laggy for me 😕

File: src/Grocery/Grocery.js [hunk 2]
```diff
@@ -1,34 +1,131 @@
...
 		return (
-			<CheckBox 
-				title={title}
-				checkedColor='green'
-				checked={item.checked}
-				onPress={() => this.handlePress()}
-				onIconPress={() => this.handlePress()}/>
-		);
+			<SafeAreaView style={{ flex: 1 }}>
+				<TextInput
```
Comment (indirap) [inline src/Grocery/Grocery.js:132]: I tried using react-native-element's `Input` component here, but the UI would always crash when I tried it. Do you know why?

Comment (zackva) [reply src/Grocery/Grocery.js]: That's weird. I've never seen that before

File: src/Grocery/Grocery.js (hunk 1 above)
Comment (zackva) [reply src/Grocery/Grocery.js]: It's working fine for me on my Mac

File: src/Grocery/Grocery.js [hunk 3]
```diff
@@ -1,34 +1,131 @@
...
+		if (this.state.name.length === 0) {
+			Alert.alert(
+				'Error updating Grocery item',
+				'Grocery name must not be empty',
+				[
+					{ text: "OK", style: 'normal'}
+				],
+				{ cancelable: true }
+			);
+		}
```
Comment (vatthikorn) [inline src/Grocery/Grocery.js:45]: Probably should bail here as now when you're adding a grocery item with empty name, it first shows this alert then tries to call `groceries.insert` below, which then errors out and shows another alert on top of the first one

File: src/Grocery/Grocery.js [hunk 4]
```diff
@@ -1,34 +1,131 @@
...
+				);
+				return;
+			}
+
+			Meteor.call('grocerylists.addItem', this.state.listId, groceryId);
+			this.props.navigation.goBack();
+		});
+	}
+
+	updateGrocery() {
```
Comment (zackva) [inline src/Grocery/Grocery.js]: Updating amount doesn't appear to work here. I updated the amount and it rewinds back to the previous screen but the cell still has the old amount/name shown

File: src/Grocery/Grocery.js [hunk 5]
```diff
@@ -1,34 +1,131 @@
...
+		if (this.state.name.length === 0) {
+			Alert.alert(
+				'Error updating Grocery item',
+				'Grocery name must not be empty',
+				[
+					{ text: "OK", style: 'normal'}
+				],
+				{ cancelable: true }
+			);
```
Comment (vatthikorn) [inline src/Grocery/Grocery.js:74]: Should return here as well as this is an error

**Review by vatthikorn — CHANGES_REQUESTED** [review]

**Comment by vatthikorn** [comment]
I'm also seeing two buttons for adding: one on the right bar button and another as a floating icon on the bottom right. The top right one should be removed to make it consistent across the app. 

Also is deleting the grocery item currently not supported yet?

File: src/Grocery/Grocery.js (hunk 1 above)
Comment (vatthikorn) [reply src/Grocery/Grocery.js]: It's working fine on my Mac...

File: src/Grocery/Grocery.js (hunk 2 above)
Comment (vatthikorn) [reply src/Grocery/Grocery.js:132]: That's weird. I've never seen that before.

**Comment by indirap** [comment]
@zackva deleting grocery items is currently not supported yet. To be honest I didn't even think of that... I'll do that as part of a separate pull request

**Comment by indirap** [comment]
@zackva I just added the ability to delete grocery items

**Review by vatthikorn — APPROVED** [review]

**Pull request merged**